detector.go

```diff
@@ -56,27 +58,30 @@ type Detector struct {
 	DirStore       DirStore
 	Executor       buildpack.DetectExecutor
 	HasExtensions  bool
-	Logger         log.Logger
+	Logger         log.LoggerWriterWithLevel
+	MultiLogger    *log.MultiLogger
 	Order          buildpack.Order
 	PlatformDir    string
 	Resolver       DetectResolver
 	Runs           *sync.Map
 }
 
-func (f *DetectorFactory) NewDetector(appDir, buildConfigDir, orderPath, platformDir string, logger log.Logger) (*Detector, error) {
+func (f *DetectorFactory) NewDetector(appDir, buildConfigDir, orderPath, platformDir string, logger log.LoggerWriterWithLevel) (*Detector, error) {
 	detector := &Detector{
 		AppDir:         appDir,
 		BuildConfigDir: buildConfigDir,
 		DirStore:       f.dirStore,
 		Executor:       &buildpack.DefaultDetectExecutor{},
 		Logger:         logger,
 		PlatformDir:    platformDir,
-		Resolver:       &DefaultResolver{Logger: logger},
 		Runs:           &sync.Map{},
 	}
 	if err := f.setOrder(detector, orderPath, logger); err != nil {
 		return nil, err
 	}
+	if err := f.setResolverAndMultiLogger(detector, logger); err != nil {
```

Comment:
What if we used the in-memory logger and then re-log using the passed in logger?

```golang
// setup in-memory logger for detect so we can dynamically show logs after execution
memLogger := memory.New()

// use that in our Detector
Resolver: &DefaultResolver{Logger: memLogger}
```

Then later on we always copy the entries from the memory logger to the default logger, but can skip log lines that don't match the level.

```golang
	for _, e := range memLogger.Entries {
           if detectErr != null {
             defaultLogger.HandleLog(e)
           } else if e.Level >= Info { efaultLogger.HandleLog(e) }
        }
        return group, plan, detectErr
```

Reply:
Hmm... but would that not result in some log messages being printed out of order?

Reply:
Would it? Is `Entries` order not guaranteed?

Reply:
Ah I think I get it now... let me make a few changes

Reply:
@jabrown85 could you take another look? It's much simpler now... thanks :)